pkg/profiler/cpu/cpu.go

```diff
@@ -476,15 +472,57 @@ func (p *CPU) Run(ctx context.Context) error {
 
 	eventsChannel := make(chan []byte)
 	lostChannel := make(chan uint64)
+	requestUnwindInfoChannel := make(chan int)
 	perfBuf, err := m.InitPerfBuf("events", eventsChannel, lostChannel, 64)
 	if err != nil {
 		return fmt.Errorf("failed to init perf buffer: %w", err)
 	}
 	perfBuf.Start()
 
+	// Process BPF events.
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case receivedBytes := <-eventsChannel:
+				if len(receivedBytes) == 0 {
+					continue
+				}
+
+				payload := binary.LittleEndian.Uint64(receivedBytes)
+				// Get the 4 more significant bytes and convert to int as they are different types.
+				// On x86_64:
+				//	- unsafe.Sizeof(int(0)) = 8
+				//	- unsafe.Sizeof(uint32(0)) = 4
+				pid := int(int32(payload))
+				if payload&RequestUnwindInformation == RequestUnwindInformation && p.DwarfUnwindingWithoutPolling() {
+					requestUnwindInfoChannel <- pid
+				} else if payload&RequestProcessMappings == RequestProcessMappings {
+					// Populate mappings cache.
+					_, _ = p.processMappings.PopulateMappings(pid)
+					// TODO: even if we populate the mappings for the process we just got
```

Comment:
I think this a perfectly acceptable trade-off. If this is truly common and important, I think one of two things would happen:

* The user uploads debuginfos themselves
* The process is spawned often enough that we eventually catch it

Reply:
Right! That's what I thought too. For this iteration we can leave it as-is and then reevaluate things once we get feedback from our users